### PR TITLE
(BOLT-518) Fact task incorrect OS Family

### PR DESCRIPTION
**Pull request by donoghuc**
Rastasheep/ubuntu-sshd no long comes with lsb_release. Use /etc/os-release to parse ubuntu facts.

### CODE REVIEW

File: modules/facts/tasks/bash.sh [hunk 1]
```diff
@@ -29,6 +29,17 @@ if [ -z "${name}" ]; then
     fi
 fi
 
+# if lsb not available try os-release
+if [ -z "${name}" ]; then
+    if [ -e /etc/os-release ]; then
+        name=$(grep "^NAME" /etc/os-release | cut -d'=' -f2 | sed "s/\"//g")
```
Comment (nicklewis) [inline modules/facts/tasks/bash.sh]: This returns `CentOS Linux` on CentOS.

Comment (nicklewis) [reply modules/facts/tasks/bash.sh]: Oh never mind, I see that we check `/etc/redhat-release` first.

Comment (donoghuc) [reply modules/facts/tasks/bash.sh]: Running the whole script? Or just the new logic bit? I tried it on a docker instance ``` docker run -ti centos:latest ``` and there was /etc/redhat-release so the script set name to CentOS. 
```
# Determine the OS name
if [ -f /etc/redhat-release ]; then
    if egrep -iq centos /etc/redhat-release; then
        name=CentOS
    elif egrep -iq 'Fedora release' /etc/redhat-release; then
        name=Fedora
    fi
    release=$(sed -r -e 's/^.* release ([0-9]+(\.[0-9]+)?).*$/\1/' \
                  /etc/redhat-release)
fi
```

Comment (donoghuc) [reply modules/facts/tasks/bash.sh]: bash.sh tested on following docker images:
```
ubuntu
fedora
centos
debian
registry.access.redhat.com/rhel7/rhel
```
## results
### ubuntu
```
ubuntu_1  | {
ubuntu_1  |   "os": {
ubuntu_1  |     "name": "Ubuntu",
ubuntu_1  |     "release": {
ubuntu_1  |       "full": "18.04",
ubuntu_1  |       "major": "18",
ubuntu_1  |       "minor": "04"
ubuntu_1  |     },
ubuntu_1  |     "family": "Debian"
ubuntu_1  |   }
ubuntu_1  | }
```
### centos
```
centos_1  | {
centos_1  |   "os": {
centos_1  |     "name": "CentOS",
centos_1  |     "release": {
centos_1  |       "full": "7.4",
centos_1  |       "major": "7",
centos_1  |       "minor": "4"
centos_1  |     },
centos_1  |     "family": "RedHat"
centos_1  |   }
centos_1  | }
```
### rhel7
```
rhel_1    | {
rhel_1    |   "os": {
rhel_1    |     "name": "Red Hat Enterprise Linux Server",
rhel_1    |     "release": {
rhel_1    |       "full": "7.5",
rhel_1    |       "major": "7",
rhel_1    |       "minor": "5"
rhel_1    |     },
rhel_1    |     "family": "Red Hat Enterprise Linux Server"
rhel_1    |   }
rhel_1    | }
```
### fedora
```
fedora_1  | {
fedora_1  |   "os": {
fedora_1  |     "name": "Fedora",
fedora_1  |     "release": {
fedora_1  |       "full": "28",
fedora_1  |       "major": "28",
fedora_1  |       "minor": ""
fedora_1  |     },
fedora_1  |     "family": "RedHat"
fedora_1  |   }
fedora_1  | }
```
### debian
```
debian_1  | {
debian_1  |   "os": {
debian_1  |     "name": "Debian GNU/Linux",
debian_1  |     "release": {
debian_1  |       "full": "9",
debian_1  |       "major": "9",
debian_1  |       "minor": ""
debian_1  |     },
debian_1  |     "family": "Debian GNU/Linux"
debian_1  |   }
debian_1  | }
```
It seems that ideally there will be two possible values for ```family``` either ```RedHat``` or ```Debian```
Should we add more logic to bash.sh to set ```family``` appropriately for the debian and rhel7 docker images?

Comment (MikaelSmith) [reply modules/facts/tasks/bash.sh]: That is the ideal for family. Is that what's happening at line 45?

**Pull request merged**